### PR TITLE
Fix the repo and test in Travis using Docker

### PR DESCRIPTION
**Pull request by simonjbeaumont**
Would be nice to go back to testing just the packages changed but at least we now have something that builds.

This also builds all the packages at once so won't weed out missing dependencies. However, we are testing that all the libraries are mutually compatible.

Given we have a container we should probably install each package in a new container to check the deps. Possibly in addition to testing them all.

### CODE REVIEW

**Comment by johnelse** [comment]
Looks sensible.

There's a typo in the commit message of c8bc57e - `s/FRONTENT/FRONTEND/`

**Comment by simonjbeaumont** [comment]
@johnelse are you saying that, if I fix that commit message, we can take this given it leaves us in no worse a position than we're currently in. Some ongoing work to make it actually pass?

File: packages/xapi-rrdd/xapi-rrdd.xapi-project#master/opam [hunk 1]
```diff
@@ -8,18 +8,18 @@ remove: [make "uninstall" "BINDIR=%{bin}%" "SBINDIR=%{bin}%" "LIBEXECDIR=%{bin}%
...
-  "xapi-backtrace"
-  "xapi-idl" {>= "0.9.21"}
+  "xapi-backtrace" {>= "0.3"}
+  "xapi-idl" {= "xapi-project#master"}
   "xapi-libs-transitional" {>= "0.9.7"}
-  "xapi-inventory"
+  "xapi-inventory" {= "xapi-project#master"}
   "xapi-stdext" {>= "0.13.0"}
   "xapi-forkexecd"
   "xenctrl" {>= "0.9.28"}
```
Comment (johnelse) [inline packages/xapi-rrdd/xapi-rrdd.xapi-project#master/opam]: This should be pinned to `xapi-project#master`

File: packages/xapi-libs-transitional/xapi-libs-transitional.xapi-project#master/opam [hunk 2]
```diff
@@ -22,11 +22,11 @@ remove: [
 os: ["linux"]
 depends: [
   "ocamlfind"
-  "xapi-stdext" {>= "0.13.0"} 
+  "xapi-stdext" {= "xapi-project#master"}
   "xmlm"
   "xapi-forkexecd"
```
Comment (johnelse) [inline packages/xapi-libs-transitional/xapi-libs-transitional.xapi-project#master/opam]: forkexecd needs adding to the repo and pinning to `xapi-project#master`

File: packages/xapi-rrdd-plugin/xapi-rrdd-plugin.xapi-project#master/opam [hunk 3]
```diff
@@ -8,8 +8,8 @@ remove: [make "PREFIX=%{prefix}%" "uninstall"]
 depends: [
   "rpc"
   "xapi-forkexecd"
```
Comment (johnelse) [inline packages/xapi-rrdd-plugin/xapi-rrdd-plugin.xapi-project#master/opam]: forkexecd needs adding to the repo and pinning to xapi-project#master

File: packages/xapi-tapctl/xapi-tapctl.xapi-project#master/opam [hunk 4]
```diff
@@ -10,6 +10,6 @@ remove: [
 ]
 depends: [
   "ocamlfind"
-  "xapi-stdext"
+  "xapi-stdext" {= "xapi-project#master"}
   "xapi-forkexecd"
```
Comment (johnelse) [inline packages/xapi-tapctl/xapi-tapctl.xapi-project#master/opam]: forkexecd needs pinning

File: packages/xapi-xenopsd/xapi-xenopsd.xapi-project#master/opam [hunk 5]
```diff
@@ -17,17 +17,17 @@ depends: [
   "uri"
   "re"
   "rpc"
-  "xapi-idl" {>= "0.10.0"}
+  "xapi-idl" {= "xapi-project#master"}
   "xapi-forkexecd"
```
Comment (johnelse) [inline packages/xapi-xenopsd/xapi-xenopsd.xapi-project#master/opam]: forkexecd needs pinning

File: packages/xapi/xapi.xapi-project#master/opam [hunk 6]
```diff
@@ -8,28 +8,28 @@ build: [
...
-  "xen-api-client"
+  "xapi-idl" {= "xapi-project#master"}
+  "xapi-libs-transitional" {= "xapi-project#master"}
+  "xen-api-client" {= "xapi-project#master"}
   "xapi-netdev"
-  "omake"
+  "omake" {build}
   "cdrom"
   "fd-send-recv"
   "xapi-forkexecd"
```
Comment (johnelse) [inline packages/xapi/xapi.xapi-project#master/opam]: forkexecd needs pinning

**Comment by johnelse** [comment]
Looks good!

**Comment by jonludlam** [comment]
Superseded by #117

**Pull request merged**